src/shared/Core/EnvironmentBase.cs

```diff
@@ -104,7 +104,12 @@ public virtual Process CreateProcess(string path, string args, bool useShellExec
             return new Process { StartInfo = psi };
         }
 
-        public bool TryLocateExecutable(string program, out string path)
+        public virtual bool TryLocateExecutable(string program, out string path)
+        {
+            return TryLocateExecutable(program, new List<string>(), out path);
```

Comment:
Pass `null` here instead of creating a new heap object

```suggestion
            return TryLocateExecutable(program, null, out path);
```